library/common/config_template.h

```diff
@@ -0,0 +1,46 @@
+#pragma once
+
+// NOLINT(namespace-envoy)
+
+/**
+ * Template configuration compiled with the Envoy Mobile library.
+ * More information about Envoy's config can be found at:
+ * https://www.envoyproxy.io/docs/envoy/latest/configuration/configuration
+ */
+extern const char* config_template;
+
+/**
+ * Template configuration used for dynamic creation of the platform-bridged filter chain.
+ */
+extern const char* platform_filter_template;
+
+/**
+ * Template configuration used for dynamic creation of the native filter chain.
+ */
+extern const char* native_filter_template;
+
+/**
+ * Template that enables the route cache reset filter in the chain.
+ * Should only be added when the route cache should be cleared on every request
+ * going through the filter chain between initial route resolution and the router
+ * filter's invocation on the request path. Typically only used for enabling
+ * direct responses to mutate headers which are then later used for routing.
+ */
+extern const char* route_cache_reset_filter_template;
```

Comment:
maybe we should move the templates that are intended for test usage to a different header.

Reply:
same as above

Reply:
To expand on this - a number of these templates are going away entirely. Interpolation itself will be going away entirely.